api/provider/tiktok.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"context"
+	"strings"
+
+	"github.com/netlify/gotrue/conf"
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultTikTokAPIBase = "https://open-api.tiktok.com/"
+)
+
+type tiktokProvider struct {
+	*oauth2.Config
+	APIPath string
+}
+
+type tiktokUser struct {
+	ID             string `json:"open_id"`
+	UnionID        string `json:"union_id"`
+	DisplayName    string `json:"display_name"`
+	AvatarUrl      string `json:"avatar_url"`
+	AvatarUrl100   string `json:"avatar_url_100"`
+	AvatarUrl200   string `json:"avatar_url_200"`
+	AvatarUrlLarge string `json:"avatar_large_url"`
+}
+
+// NewTikTokProvider creates a TikTok account provider.
+func NewTikTokProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuthProvider, error) {
+	if err := ext.Validate(); err != nil {
+		return nil, err
+	}
+
+	apiPath := chooseHost(ext.URL, defaultTikTokAPIBase)
+
+	oauthScopes := []string{
+		"user.info.basic",
+	}
+
+	if scopes != "" {
+		oauthScopes = append(oauthScopes, strings.Split(scopes, ",")...)
+	}
+
+	return &tiktokProvider{
+		Config: &oauth2.Config{
+			ClientID:     ext.ClientID,
+			ClientSecret: ext.Secret,
+			Endpoint: oauth2.Endpoint{
+				AuthURL:  apiPath + "/platform/oauth/connect",
+				TokenURL: apiPath + "/oauth/access_token/",
+			},
+			Scopes:      oauthScopes,
+			RedirectURL: ext.RedirectURI,
+		},
+		APIPath: apiPath,
+	}, nil
+}
+
+func (g tiktokProvider) GetOAuthToken(code string) (*oauth2.Token, error) {
+	return g.Exchange(oauth2.NoContext, code)
+}
+
+func (g tiktokProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*UserProvidedData, error) {
+	var u tiktokUser
+	if err := makeRequest(ctx, tok, g.Config, g.APIPath+"/user/info", &u); err != nil {
```

Comment:
This needs to be a post request with JSON body as per the docs - https://developers.tiktok.com/doc/login-kit-user-info-basic. It doesn't appear that `makeRequest` supports that, and some custom request logic may be needed as the ID from the token response is also necessary.

\\cc @kangmingtay

Reply:
Fixed with the commit https://github.com/supabase/gotrue/pull/269/commits/c47b9615e5c4a5a80a02743a665f25335f10171c